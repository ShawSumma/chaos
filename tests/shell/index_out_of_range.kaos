array a = [1,2,3]
a[0]
a[1]
a[2]
a[3]
a[123400]
a[-1]
a[-2]
a[-3]
a[-4]
a[-123400]

str b = "mike"
b[0]
b[1]
b[2]
b[3]
b[4]
b[123400]
b[-1]
b[-2]
b[-3]
b[-4]
b[-5]
b[-123400]
