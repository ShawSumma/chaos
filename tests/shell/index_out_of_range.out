1
2
3
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 5, in <module>
Index out of range: 3 for list: a
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 1, in <module>
Index out of range: 123400 for list: a
Absorbed by Interactive Shell
3
2
1
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 4, in <module>
Index out of range: -4 for list: a
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 1, in <module>
Index out of range: -123400 for list: a
Absorbed by Interactive Shell
m
i
k
e
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 7, in <module>
Index out of range: 4 for string: b
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 1, in <module>
Index out of range: 123400 for string: b
Absorbed by Interactive Shell
e
k
i
m
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 5, in <module>
Index out of range: -5 for string: b
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 1, in <module>
Index out of range: -123400 for string: b
Absorbed by Interactive Shell
[]
[start] =>
	{id: 0, name: a, 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 4, 2nd_type: 3, value_type: 4, role: 0, param_of: , value: [1, 2, 3]} =>
	{id: 1, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: , value: 3} =>
	{id: 2, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: , value: 2} =>
	{id: 3, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: , value: 1} =>
	{id: 14, name: b, 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: mike} =>
	{id: 16, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: m} =>
	{id: 18, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: i} =>
	{id: 20, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: k} =>
	{id: 22, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: e} =>
	{id: 26, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: e} =>
	{id: 28, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: k} =>
	{id: 30, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: i} =>
	{id: 32, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: m} =>
	{id: 35, name: f, 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 4, 2nd_type: 0, value_type: 4, role: 0, param_of: , value: []} =>
[end]
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 7, in <module>
The variable name 'f' is already defined!
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 1, in <module>
Index out of range: 0 for list: f
Absorbed by Interactive Shell
[]
[start] =>
	{id: 0, name: a, 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 4, 2nd_type: 3, value_type: 4, role: 0, param_of: , value: [1, 2, 3]} =>
	{id: 1, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: , value: 3} =>
	{id: 2, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: , value: 2} =>
	{id: 3, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: , value: 1} =>
	{id: 14, name: b, 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: mike} =>
	{id: 16, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: m} =>
	{id: 18, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: i} =>
	{id: 20, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: k} =>
	{id: 22, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: e} =>
	{id: 26, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: e} =>
	{id: 28, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: k} =>
	{id: 30, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: i} =>
	{id: 32, name: (null), 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: , value: m} =>
	{id: 35, name: f, 2nd_name: (null), key: (null), scope: N/A, depth: 0, type: 4, 2nd_type: 0, value_type: 4, role: 0, param_of: , value: []} =>
[end]
[1, 2]
Chaos Error:
Module: ~/chaos/__interactive__.kaos", line 8, in <module>
Index out of range: 2 for list: a
Absorbed by Interactive Shell
